### PR TITLE
Clicky/katalyst deployer

### PR DESCRIPTION
**Pull request by Anyhowclick**
- Deployer script for TradeLogic, Network, Proxy, FeeHandler contracts
- Minor changes to KN.sol and KNP.sol
- Use latest solc version (not really needed)
- Made compileContracts.js to work for both v4 and v5 contracts
- Modified APR deployer script too

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "rlp": "2.2.4",
     "secp256k1": "3.8.0",
     "sha3": "2.1.1",
-    "solc": "0.5.11",
+    "solc": "0.6.3",
```
Comment (ilanDoron) [inline package.json]: why 0.6.3?

File: web3deployment/liquidityReserve_input.json [hunk 2]
```diff
@@ -1,25 +1,29 @@
 {
-  "tokens": {
-    "KNC": {
+  "token": {
       "name": "KyberNetwork",
+      "symbol": "KNC",
       "decimals": 18,
-      "address": "0x4E470dc7321E84CA96FcAEDD0C8aBCebbAEB68C6"
-    }
+      "address": "0xa67bba948bd8372bc9e60b1acfafb8e2db3cc57c"
```
Comment (ilanDoron) [inline web3deployment/liquidityReserve_input.json]: is this for ropsten?
we should organize this folder.
All ropsten stuff should move to ropsten folder.
we don't want someone using this thinking its main net KNC
and this folder is not very cluttered.

File: package.json (hunk 1 above)
Comment (Anyhowclick) [reply package.json]: as discussed, will fix if 0.5.11 suffices

File: web3deployment/liquidityReserve_input.json (hunk 2 above)
Comment (Anyhowclick) [reply web3deployment/liquidityReserve_input.json]: Changing it back to the ropsten KNC address. For deployment, project teams will be using replacing the token field with their own token.

Agree that we should organize the folder

Comment (Anyhowclick) [reply web3deployment/liquidityReserve_input.json]: I leave it as is first, we can re-org the folder in another PR =)

File: web3deployment/KatalystDeploy/compileContractsV5.js [hunk 3]
```diff
@@ -1,94 +0,0 @@
...
-    // console.log(config)
-    let output = compileSources(config);
-    errorHandling(output);
-    // console.log(output.contracts['GasHelper.sol']);
-    // console.log(output);
-
-    return output;
-}
-
-main();
```
Comment (ilanDoron) [inline web3deployment/KatalystDeploy/compileContractsV5.js]: 1 more thing I added here which I found helpful.
if contract is called with no script it just compiles. and better if it also output result.
so if anyone updates the compile contracts file. he can run it without needing any script for checking the update is working as expected.
So I suggest each compile contracts file will have 1 export function
and main function that calls the compile function and prints output.errors.

File: web3deployment/compileContracts.js [hunk 4]
```diff
@@ -1,55 +1,161 @@
...
+        setTimeout(resolve,ms)
+    })
+}
+
+module.exports.compileContracts = compileContracts;
+
+async function compileContracts(versionNum) {
+    let solcVersionNum;
+    let sourceFiles;
+    if (versionNum == "v4") {
```
Comment (ilanDoron) [inline web3deployment/compileContracts.js]: its great u could work this out!!

File: web3deployment/KatalystDeploy/compileContractsV5.js [hunk 5]
```diff
@@ -1,94 +0,0 @@
-const fs = require('fs-extra');
```
[the remark this answers is not in the document]
Comment (Anyhowclick) [reply web3deployment/KatalystDeploy/compileContractsV5.js]: compileContracts.js allows for either v4 or v5. I tried modifying to allow both, but the async await messes with the compiler variable. What I mean is that if we compile v5, then v4, the v4 compiler doesn't download and load up in time for compilation, so I get an error.....

I've not found a good fix for it yet... Nice to have, medium priority at this time? All u need to do is call the the function twice! =)

File: web3deployment/KatalystDeploy/katalystNetworkDeployer.js [hunk 6]
```diff
@@ -0,0 +1,526 @@
+#!/usr/bin/env node
+
```
[the remark this answers is not in the document]
Comment (Anyhowclick) [reply web3deployment/KatalystDeploy/katalystNetworkDeployer.js]: Ah.. so 
Stage 1: Deploy contracts, save as json.
Stage 2: Setup contracts

The functions are modularized enough to allow this =)

**Pull request merged**